tests/unittests/test_refresh_access_token.py

```diff
@@ -0,0 +1,158 @@
+import unittest
+from unittest.mock import MagicMock, patch
+from datetime import datetime
+from singer import utils
+from tap_square.client import require_new_access_token, SquareClient
+
+REFRESH_TOKEN_BEFORE = 22
+
+
+class TestRequireNewAccessToken(unittest.TestCase):
+    def setUp(self):
+        self.client = MagicMock()
+        self.access_token = "test_access_token"
+        self.token_expiry = "2025-01-09T12:23:41Z"
+
+    def test_no_access_token(self):
+        """Return true when no access token is provided."""
+        result = require_new_access_token(None, self.client)
+        self.assertTrue(result)
+
+    @patch("singer.http_request_timer")
+    @patch("singer.utils.now")
+    def test_valid_access_token(self, mock_now, mock_timer):
+        """Return false when the token generated in less than 7 days."""
+        mock_now.return_value = utils.strptime_with_tz("2024-12-14T12:23:41Z")
+        self.client.o_auth.retrieve_token_status.return_value = MagicMock(
+            is_error=MagicMock(return_value=False),
+            body={"expires_at": self.token_expiry},
+        )
+        result = require_new_access_token(self.access_token, self.client)
+        self.assertFalse(result)
+
+    @patch("singer.http_request_timer")
+    @patch("singer.utils.now")
+    def test_expired_access_token(self, mock_now, mock_timer):
+        """Return true when the token is expired."""
+        mock_now.return_value = utils.strptime_with_tz("2024-12-24T12:23:41Z")
```

Comment:
Instead of hard-coded dates we should use relative dates to current dates.

Reply:
Done with unit tests update